crates/database/src/auction_prices.rs

```diff
@@ -0,0 +1,81 @@
+use {
+    crate::{auction::AuctionId, Address},
+    bigdecimal::BigDecimal,
+    sqlx::{Connection, PgConnection},
+};
+
+/// External token price for a given auction.
+#[derive(Debug, Clone, PartialEq, sqlx::FromRow)]
+pub struct AuctionPrice {
+    pub auction_id: AuctionId,
+    pub token: Address,
+    pub price: BigDecimal,
+}
+
+pub async fn insert(ex: &mut PgConnection, prices: Vec<AuctionPrice>) -> Result<(), sqlx::Error> {
+    let mut transaction = ex.begin().await?;
+    const QUERY: &str =
+        r#"INSERT INTO auction_prices (auction_id, token, price) VALUES ($1, $2, $3);"#;
+    for price in prices {
+        sqlx::query(QUERY)
+            .bind(price.auction_id)
+            .bind(price.token)
+            .bind(price.price)
+            .execute(&mut *transaction)
+            .await?;
+    }
+    transaction.commit().await?;
+    Ok(())
+}
+
+pub async fn delete(ex: &mut PgConnection, auction_id: AuctionId) -> Result<(), sqlx::Error> {
+    const QUERY: &str = "DELETE FROM auction_prices WHERE auction_id = $1";
```

Comment:
This table is missing a primary key on `(auction_id, token)` because it does not make sense to have duplicates and it is important for the efficiency of the query.

Reply:
Addressed in https://github.com/cowprotocol/services/pull/1271